src/issue.ts

```diff
@@ -0,0 +1,105 @@
+    this.similarity = +core.getInput("similarity", {required: false})
+    this.bodyValue = +core.getInput("body-value", {required: false})
+  }
+
+  public determineArea(): string {
+    let titleIssueWords = this.title.split(/ |\./);
+    let bodyIssueWords = this.body.split(/ |\./)
+    let titleValue: number = 1
+    let x: number = 1;
+    let potentialAreas: Map<string, number> = new Map()
+      
+    // For each word in the title, check if it matches any keywords. If it does, add decreasing score based on inverse function to the area keyword is in.
+    titleIssueWords.forEach(content => {
+      potentialAreas = this.scoreArea(content, potentialAreas, titleValue);
+      ++x
+      titleValue = (2/(1+x))
+    })
+      
+    // Add static value to area keyword is in if keyword is found in body
+    bodyIssueWords.forEach(content => {
+      potentialAreas = this.scoreArea(content, potentialAreas, this.bodyValue);
+    })
+      
+    console.log(...potentialAreas)
+
+    let winningArea = '';
+    let winners: Map<string,number> = new Map();
+    for (let area of potentialAreas.entries()) {
+      if(winners.size === 0) {
+        winners.set(area[0], area[1]);
+      } else if (area[1] > winners.values().next().value) {
+        winners = new Map();
+        winners.set(area[0], area[1]);
+      } else if (area[1] === winners.values().next().value) {
+        winners.set(area[0], area[1]);
+      }
+    }
+    // tiebreaker goes to the area with more *exact* keyword matches
+    if(winners.size > 1 && this.similarity !== 0) {
+      winningArea = this.determineArea();
+    } else if (winners.size > 0) {
+      winningArea = winners.keys().next().value;
+    } 
+      
+    winningArea = winners.keys().next().value;
+      
+    return winningArea;
+  }
+
+  private scoreArea(content: string, potentialAreas: Map<string, number>, value): Map<string, number> {
+    this.parameters.forEach(obj => {
+      obj.keywords.forEach(keyword => {
+        if(this.similarStrings(content, keyword)) {
+          potentialAreas.has(obj.area) ?
+            potentialAreas.set(obj.area, potentialAreas.get(obj.area)+value) :
+            potentialAreas.set(obj.area, value);
+        }    
+      })
+    })    
+    return potentialAreas;
+  }
+
+  private similarStrings(str1: string, str2: string): boolean {
+    str1 = str1.toLowerCase();
+    str2 = str2.toLowerCase();
+        
+    // Regex for removing punctuation and replacing with empty string
+    str1 = str1.replace(/ |_|-|\(|\)|:|\[|\]|	|\./gi, '');
+    str2 = str2.replace(/ |_|-|\(|\)|:|\[|\]|	|\./gi, '');
+      
+    // levenshtein returns a value between 0 and the length of the strings being compared. This
+    // represents the number of character differences between compared strings. We compare this
+    // with a set percentage of the average length of said strings
+    if(levenshtein(str1, str2) <= ((str1.length + str2.length) / 2) * this.similarity)
```

Comment:
Same comment as magic numbers above. While this is less of a magic number issue, it still makes sense to separate the calculation from the condition. Something like the following:

```ts
const isSimilar: number = (len1: number, len2: number, sim: number) => {
	return ((len1 + len2) / 2) * sim;
};

...

if(levenshtein(str1, str2) <= isSimilar(str1.length, str2.length, this.similarity))

...
```

While its not much simpler inline, it separates the logic